sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -35,6 +35,8 @@ public class ChatClient {
     private let credential: CommunicationTokenCredential
     private let options: AzureCommunicationChatClientOptions
     private let service: Chat
+    private let signallingClient: CommunicationSignallingClient?
```

Comment:
good point... if user doesn't enable real-time notification, do we need to initialize the SignallingClient? If not, we should move the initialization to after user calls startRealTimeNotification? If we initialize it regardless, then it doesn't need to be optional.

Reply:
I thought it was optional because if get we get an error accessing the token [(line73)](https://github.com/knvsl/azure-sdk-for-ios/blob/7b01b92d72774434b20a91cdc3b8d128e5d04c11/sdk/communication/AzureCommunicationChat/Source/ChatClient.swift#L73) then signalling client cannot be initialized.

But if it is optional to avoid unneccessary work given the user never enables notifications, then we can move the init into `startRealtTimeNotifications()` & throw an exception from there if there's token errors. The downside to me is that there is no fallback for the user to re-init the signalling client except creating a new `ChatClient`, which is why I would lean towards throwing from `ChatClient`

Reply:
^not actually sure what would cause token errors though, if this is even a scenario that is expected?